### DISCUSSION TITLE
Feature request: add Content-Security-Policy headers

### DISCUSSION THREAD

**Discussion by basbebe** [modoboa/modoboa#1414] [Ideas]
Modoboa should be able to send it's own `Content-Security-Policy` HTTP headers.
It script-src only allowed specific, dynamically created nonces, security could be enhanced for supporting browsers.

**Comment by kryskool** [comment]
Hi @bax- 

CSP can be configure in Nginx or Apache to inject the proper header, 

I prefer this solution, because you can modify template, and use Google analytic or new theme and font, that required to adjust the CSP HTTP header.

Regards,

**Comment by ghost** [comment]
I've just done a little testing and at the moment Modoboa (and it's 3rd party dependencies)  make use of inline scripts, css and the javascript eval function (ckeditor), so you would need `unsafe-inline` and `unsafe-eval` in the CSP header which removes most of the benefit of this header.

It's defiantly something to look at with the UI redesign #1320
